presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -484,6 +491,7 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             // TODO: implement full properties for local exchanges
             if (node.getScope() == LOCAL) {
                 ActualProperties.Builder builder = ActualProperties.builder();
+                builder.local(localProperties.build());
```

Comment:
What does it mean for local exchange to preserve local sorting properties? How can it be possible?

Reply:
Local exchange that has ordering scheme is implicitly of type `GATHERING` and translates into `LocalMergeOperator` that does preserve ordering.